tools/cli-bootstrap/cli.bootstrap.py

```diff
@@ -0,0 +1,562 @@
+
+            RoverPrint(RoverMods.Blue('has finished patching %s'%(RoverMods.Yellow(patchTarget_folder))))
+    except:
+        RoverSettings._DevMode = True
+        UnexpectedRoverException(sys.exc_info())
+
+##
+## END ROVER FUNCTION DEFINITIONS
+##
+
+if __name__ == "__main__":
+    ##
+    ##  COMMAND-LINE BEHAVIOR
+    ##
+
+    parser = argparse.ArgumentParser(description = 'Rover is the dotnet bootstrapping tool.')
+    parser.add_argument('-clone', metavar='c', nargs='*', default=['coreclr', 'corefx', 'core-setup', 'libuv'], help='Clones specified repositories in to the working directory. Select from the following repositories: {' 
+        + '%s, %s, %s, %s'%(RoverMods.Red('coreclr'), RoverMods.Blue('corefx'), RoverMods.Green('core-setup'), RoverMods.Yellow('libuv') +'}'))
+    parser.add_argument('-build', metavar='b', nargs='*', default = ['coreclr', 'corefx', 'core-setup', 'libuv'],help='\'Builds\' all native components if no arguments are specified. Otherwise, specify one or more (space separated) arguments from the following : {' 
+        + '%s, %s, %s, %s'%(RoverMods.Red('coreclr'), RoverMods.Blue('corefx'), RoverMods.Green('core-setup'), RoverMods.Yellow('libuv') +'}'))
+    parser.add_argument('-nopatch', action='store_true', default=False, help='prevents the copying of specific native binaries from the pre-built repositories in to the destination directory.')
+    parser.add_argument('-payload', nargs=1, help='Specify a path to a tarball (something that we can tar xf) that contains a version of the dotnet CLI.')
+    
+    args = parser.parse_args()
+
+    if args.payload:
+        RoverPrint('using payload ' + RoverMods.White(str(args.payload)))
+    
+    RoverPrint('Cloning Set: ' + RoverMods.White(str(args.clone)))
+    RoverPrint('Building Set: ' + RoverMods.White(str(args.build)))
+    RoverPrint('Is Patching? ' + RoverMods.White(str(not args.nopatch)))
+    
+    RoverSettings.CloneSet = args.clone
+    RoverSettings.BuildSet = args.build
+
+    # I am guessing that users are more inclined to want patching to happen whenever it can, and so I ask
+    # for specificity in the instances that they do not want patching.
+    RoverSettings.Patch = not args.nopatch 
+
+    if args.payload:
+        RoverSettings.PayloadPath = args.payload[0]
+    ## 
+    ## END COMMAND-LINE BEHAVIOR
+    ##
+
+    ##
+    ## BEGIN DECLARATIONS
+    ##
+
+    coreclr_working_git_directory       = path.join(RoverSettings._srcDirectory, 'coreclr')
+    corefx_working_git_directory        = path.join(RoverSettings._srcDirectory, 'corefx')
+    core_setup_working_git_directory    = path.join(RoverSettings._srcDirectory, 'core-setup')
+    libuv_working_git_directory         = path.join(RoverSettings._srcDirectory, 'libuv')
+
+    default_coreclr_bin_directory       = '%s/bin/Product/Linux.x64.Release/'%(coreclr_working_git_directory)
```

Comment:
Yeah, I kind of want to see the build system overall stabilize there. Isn't there an API or something coming along in this regard? Otherwise this tool will have to chase the repositories behaviors around everywhere (as it seems we do anyways).